src/openforms/contrib/objects_api/validators.py

```diff
@@ -78,3 +92,52 @@ def validate_document_type_references(
 
     if errors:
         raise ValidationError(errors)
+
+
+def validate_object_ownership(submission: Submission) -> None:
+    form = submission.form
+
+    try:
+        auth_info = submission.auth_info
+    except ObjectDoesNotExist:
+        raise PermissionDenied("Cannot pass data reference as anonymous user")
+
+    object = None
+    for backend in form.registration_backends.filter(backend="objects_api"):
```

Comment:
We really can't have hardcoded plugin IDs in "generic" code, even if it's in the right domain. Custom extensions can subclass behaviour and they would be registered with a different ID, the ID could change in the future... And then this code won't properly work.

The plugin model is deliberately designed so that we don't need `if plugin == "foo"` code and to drop all the implementation details for a specific plugin in that plugin code. The only place where you can target a particular ID, is in data migrations.

On top of that - the initial data reference mechanism is generic - it could (and will at some point) also apply to ZGW API's. So I really expect a pattern along the lines of:

```
# during (pre?)-registration and prefill
if submission.initial_data_reference:
    plugin = lookup_plugin(submission)
    plugin.verify_ownership(submission)
```

and then the `plugin.verify_ownership` implementation can make use of the helpers in the shared functionality, but ultimately, when you have the plugin (registration) or variable (prefill) reference/instance, you have access to the necessary configuration:

* the API group to use
* the objecttype and its version
* the configured path/location that stores the identifier in the object

Architecture wise, everything in Open Forms works by:

1. invoke module (auth, prefill, registration)
2. resolve relevant plugin from form configuration + module
3. defer implementation to plugin instance

So, I'd expect helper functions in this package with the signature:

```py
def verify_ownership(
    submission: Submission, 
    client: ObjectsClient, 
    object_attribute: list[str],  # or a glom.Path
):
    ...
```

the `get_objects_client` context manager would be invoked somewhere in the caller, so that you can properly optimize connection pooling.

Reply:
I've modified it to create the client in the respective plugins